documentation/seed-node.md

```diff
@@ -0,0 +1,255 @@
+
+`make install`
+
+If we want to use Postgres as our backing datastore, we need to use the vulcanize fork of go-ipfs. This fork supports
+the Postgres datastore plugin and has been adjusted to use `dep` instead of Go modules since Go modules cannot work with
+un-versioned forks and we need to use an un-versioned fork of go-ipfs-config.
+
+Start by adding the fork and switching over to it:
+
+`git remote add vulcanize https://github.com/vulcanize/go-ipfs.git`
+
+`git fetch vulcanize`
+
+`git checkout -b postgres_update vulcanize/postgres_update`
+
+Now install this fork of ipfs, first be sure to remove any previous installation. It is important to use the below command
+instead of using `make install`, as `make install` will default to using Go modules and this will wreck our dependencies
+since Go modules cannot work with un-versioned forks.
+
+`go install ./cmd/ipfs`
+
+Check that is installed properly by running
+
+`ipfs`
+
+You should see the CLI info/help output.
+
+And now we initialize with the `postgresds` profile.
+If ipfs was previously initialized we will need to remove the old profile first.
+We also need to provide env variables for the postgres connection: 
+
+We can either set these manually, e.g.
+```bash
+export IPFS_PGHOST=
+export IPFS_PGUSER=
+export IPFS_PGDATABASE=
+export IPFS_PGPORT=
+export IPFS_PGPASSWORD=
+```
+
+And then run the ipfs command
+
+`ipfs init --profile=postgresds`
+
+Or we can use the pre-made script at `GOPATH/src/github.com/ipfs/go-ipfs/misc/utility/ipfs_postgres.sh`
+which has usage: 
+
+`./ipfs_postgres.sh <IPFS_PGHOST> <IPFS_PGPORT> <IPFS_PGUSER> <IPFS_PGDATABASE>"`
+
+and will ask us to enter the password, avoiding storing it to an ENV variable.
+
+Once we have initialized ipfs, that is all we need to do with it- we do not need to run a daemon during the subsequent processes.
+
+### Geth 
+For Geth, we currently *require* a special fork, and we can set this up as follows:
+
+Begin by downloading geth and switching to the vulcanize/rpc_statediffing branch
+
+`go get github.com/ethereum/go-ethereum`
+
+`cd $GOPATH/src/github.com/ethereum/go-ethereum`
+
+`git remote add vulcanize https://github.com/vulcanize/go-ethereum.git`
+
+`git fetch vulcanize`
+
+`git checkout -b rpc_statediffing vulcanize/rpc_statediffing`
+
+Now, install this fork of geth (make sure any old versions have been uninstalled/binaries removed first)
+
+`make geth`
+
+And run the output binary with statediffing turned on:
+
+`cd $GOPATH/src/github.com/ethereum/go-ethereum/build/bin`
+
+`./geth --statediff --statediff.streamblock --ws --syncmode=full`
+
+Note: other CLI options- statediff specific ones included- can be explored with `./geth help`
+
+The output from geth should mention that it is `Starting statediff service` and block synchronization should begin shortly thereafter.
+Note that until it receives a subscriber, the statediffing process does essentially nothing. Once a subscription is received, this 
```

Comment:
Good point - is this something that could be handled with the `subscription.backfill` or `subscription.backfillOnly` flags?

Reply:
Yeah the `backfill` ability allows a transformer to request data that the seed node has previously synced (seed node serves as "light" index for archival data cached on ipfs); but if the the seed node also has gaps/is missing data from the beginning of its sync with the state-diffing geth node then that data is out of reach to the transformer. 

If we have have to restart the seed node there will be a gap in the cache and there is currently no way of filling it other than going back before that height and resending with the geth node. I don't think there is anyway to refill it with a full node other than that, we would need an archival geth node fill it in otherwise. In the grand scheme of things, there would be multiple seed nodes running and publishing data to ipfs and sharing their CID indexes across the network and this redundancy would prevent/fill in any gaps.

Reply:
Yes, we just have to make sure we have the probes set up correctly.

Seed nodes should also be able to backfill from other seed nodes. I'm not sure how much of that feature we have done.